model/model/v1alpha/model.proto

```diff
@@ -1737,3 +1738,117 @@ message UndeployOrganizationModelAdminRequest {
 
 // UndeployOrganizationModelAdminResponse represents a response for a undeployed model
 message UndeployOrganizationModelAdminResponse {}
+
+// ModelTrigger contains information about trigger of models.
+message ModelTrigger {
+  // TriggerStatus defines the status of a model trigger
+  enum TriggerStatus {
+    // Unspecified
+    TRIGGER_STATUS_UNSPECIFIED = 0;
+    // model trigger in progress
+    TRIGGER_STATUS_PROCESSING = 1;
+    // model trigger succeeded
+    TRIGGER_STATUS_COMPLETED = 2;
+    // model trigger failed
+    TRIGGER_STATUS_FAILED = 3;
+    // model trigger is waiting to be executed
+    TRIGGER_STATUS_QUEUED = 4;
+  }
+
+  // TriggerSource defines the source of a model trigger
+  enum TriggerSource {
+    // Unspecified
+    TRIGGER_SOURCE_UNSPECIFIED = 0;
+    // model trigger from frontend UI
+    TRIGGER_SOURCE_CONSOLE = 1;
+    // model trigger from API or SDK
+    TRIGGER_SOURCE_API = 2;
+  }
+
+  // Model Trigger UUID.
+  string uid = 1 [(google.api.field_behavior) = OUTPUT_ONLY];
+  // Model UUID.
+  string model_uid = 2 [(google.api.field_behavior) = OUTPUT_ONLY];
+  // Pipeline Trigger UUID.
+  string trigger_uid = 3 [(google.api.field_behavior) = OUTPUT_ONLY];
+  // Model trigger status
+  TriggerStatus status = 4 [(google.api.field_behavior) = OUTPUT_ONLY];
+  // Model visibility.
+  Model.Visibility visibility = 5 [(google.api.field_behavior) = OUTPUT_ONLY];
```

Comment:
I don't think we need the visibility here. This applies to the model, not to the trigger.